### PR TITLE
Add Script of Today's Date and Time

### PR DESCRIPTION
**Pull request by utsavkhemka21**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: scripts/validate_datetime.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    """
+    subtract = datetime.timedelta(hours=b)
+    difference = a - subtract
+    return difference
+
+
+if __name__ == "__main__":
+    doctest.testmod()
+
+print
```
Comment (CaedenPH) [inline scripts/validate_datetime.py:54]:
```suggestion
```
This does nothing

File: scripts/validate_datetime.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+"""
+This is pure Python implementation of DateTime
+
+For doctests run following command:
+
+python3 validate_datetime.py
+
```
Comment (CaedenPH) [inline scripts/validate_datetime.py:7]:
```suggestion
```

File: scripts/validate_datetime.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
+"""
+This is pure Python implementation of DateTime
```
Comment (CaedenPH) [inline scripts/validate_datetime.py:2]: Could you explain in more depth please and also link some relevant documentation

File: scripts/validate_datetime.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+For doctests run following command:
+
+python3 validate_datetime.py
+
+"""
+
+
+import datetime
+import doctest
+import os
```
Comment (CaedenPH) [inline scripts/validate_datetime.py:13]:
```suggestion
```
This import is unused

File: scripts/validate_datetime.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+
+def parseOptions():
+
+    import optparse
+
+    parser = optparse.OptionParser(usage="-h")
+    parser.add_option("-d", "--difference", type="int")
+    (options, args) = parser.parse_args()
+    return options
```
Comment (CaedenPH) [inline scripts/validate_datetime.py:23]: This doesn't do much

File: scripts/validate_datetime.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+"""
+
+
+import datetime
+import doctest
+import os
+
+
+def parseOptions():
```
Comment (CaedenPH) [inline scripts/validate_datetime.py:16]:
```suggestion
def parse_options():
```
Python variable names assume `snake_case`

File: scripts/validate_datetime.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    >>> subtime(datetime.datetime(2013,11,11,11,0),24)
+    datetime.datetime(2013, 11, 10, 11, 0)
+
+    >>> subtime(datetime.datetime(2013,11,11,11,0),0)
+    datetime.datetime(2013, 11, 11, 11, 0)
+
+    >>> subtime(datetime.datetime(2013,11,11,11,0),-5)
+    datetime.datetime(2013, 11, 11, 16, 0)
+
+    """
```
Comment (CaedenPH) [inline scripts/validate_datetime.py:45]:
```suggestion
    >>> subtime(datetime.datetime(2013,11,11,11,0),10)
    datetime.datetime(2013, 11, 11, 1, 0)
    >>> subtime(datetime.datetime(2013,11,11,11,0),24)
    datetime.datetime(2013, 11, 10, 11, 0)
    >>> subtime(datetime.datetime(2013,11,11,11,0),0)
    datetime.datetime(2013, 11, 11, 11, 0)
    >>> subtime(datetime.datetime(2013,11,11,11,0),-5)
    datetime.datetime(2013, 11, 11, 16, 0)
    """
```
Takes up less lines

File: scripts/validate_datetime.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    doctest.testmod()
+
+print
+print("This is the time now -", now.strftime("%I:%M:%S %p %a, %B %d %Y"))
+difference = subtime(now, 10)
+print(
+    "This is the time minus the difference -",
+    difference.strftime("%I:%M:%S %p %a, %B %d %Y"),
+)
+print
```
Comment (CaedenPH) [inline scripts/validate_datetime.py:61]: This should be indented into the `if __name__ == "__main__"`

File: scripts/validate_datetime.py [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+    import optparse
+
+    parser = optparse.OptionParser(usage="-h")
+    parser.add_option("-d", "--difference", type="int")
+    (options, args) = parser.parse_args()
+    return options
+
+
+now = datetime.datetime.now()
```
Comment (CaedenPH) [inline scripts/validate_datetime.py:26]: Put this into the test below

File: scripts/validate_datetime.py [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+
+
+def subtime(a, b):
```
Comment (CaedenPH) [inline scripts/validate_datetime.py:29]: Please typehint the parameters and return type

File: scripts/validate_datetime.py [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+    >>> subtime(datetime.datetime(2013,11,11,11,0),24)
+    datetime.datetime(2013, 11, 10, 11, 0)
+
+    >>> subtime(datetime.datetime(2013,11,11,11,0),0)
+    datetime.datetime(2013, 11, 11, 11, 0)
+
+    >>> subtime(datetime.datetime(2013,11,11,11,0),-5)
+    datetime.datetime(2013, 11, 11, 16, 0)
+
```
Comment (CaedenPH) [inline scripts/validate_datetime.py:44]: Could you add an example of a failing doctest and also add some check within this function ensuring that the types are correct

File: scripts/validate_datetime.py (hunk 8 above)
Comment (realstealthninja) [inline scripts/validate_datetime.py:61]:
```suggestion
```

Comment (realstealthninja) [reply scripts/validate_datetime.py:61]: this does nothing either

**Pull request closed without merging**